amr-wind/utilities/tagging/CurvatureRefinement.cpp

```diff
@@ -0,0 +1,120 @@
+#include "amr-wind/utilities/tagging/CurvatureRefinement.H"
+#include "amr-wind/CFDSim.H"
+
+#include "AMReX.H"
+#include "AMReX_ParmParse.H"
+
+namespace amr_wind {
+
+CurvatureRefinement::CurvatureRefinement(const CFDSim& sim)
+    : m_sim(sim)
+    , m_curv_value(
+          m_sim.mesh().maxLevel() + 1, std::numeric_limits<amrex::Real>::max())
+{}
+
+void CurvatureRefinement::initialize(const std::string& key)
+{
+    amrex::ParmParse pp(key);
+    std::string fname;
+    pp.query("field_name", fname);
+
+    const auto& repo = m_sim.repo();
+    if (!repo.field_exists(fname)) {
+        amrex::Abort("CurvatureRefinement: Cannot find field = " + fname);
+    }
+    m_field = &(m_sim.repo().get_field(fname));
+
```

Comment:
Is this needed? Or is the field always going to be `curvature`?

Reply:
Well you need to define on which field curvature needs to be computed. I use density for vof but you may use any field you want